notation.go

```diff
@@ -215,15 +215,16 @@ func Verify(ctx context.Context, verifier Verifier, repo registry.Repository, op
 	}
 
 	// If there's no signature associated with the reference
-	if len(verificationOutcomes) == 0 {
+	if numOfSignatureProcessed == 0 {
 		return ocispec.Descriptor{}, nil, ErrorSignatureRetrievalFailed{Msg: fmt.Sprintf("no signature is associated with %q, make sure the image was signed successfully", artifactRef)}
 	}
 
-	// check whether verification was successful or not
-	if verificationOutcomes[len(verificationOutcomes)-1].Error != nil {
+	// Verification Failed
+	if len(verificationOutcomes) == 0 {
 		return ocispec.Descriptor{}, verificationOutcomes, ErrorVerificationFailed{}
```

Comment:
Looks like we don't return any verification outcomes if there is no "good" signature associated with the given reference. That means, from notation cli, a user gets zero information on why the verification failed. All they see is `signature verification failed` message, which is very generic. They don't see the exact error messages associated with the "bad" signatures in order to understand the nature of the failure and potentially update their trust store or trust policy.

Reply:
@priteshbandi @shizhMSFT
This is the result of not processing and returning the outcomes of *all* signatures.

Reply:
> Looks like we don't return any verification outcomes if there is no "good" signature associated with the given reference. That means, from notation cli, a user gets zero information on why the verification failed. All they see is `signature verification failed` message, which is very generic. They don't see the exact error messages associated with the "bad" signatures in order to understand the nature of the failure and potentially update their trust store or trust policy.

Just propose another option here:
1. Verification succeeded: return successful outcomes (there might be more than one signature succeeded) only.
2. Verification failed: still return all outcomes with details (all of them are failed).

@shizhMSFT @priteshbandi What do you think?

Reply:
My two cents:
- Limit the number of signatures attempted, for example, 50 (need to track this with an issue for later improvements)
- If one signature is verified, then the corresponding outcome should be recorded and returned. Library doesn't make any decision on what should be returned or not. The library returns the "fact". The debug option will benefit from this.

An example:
- If verification succeeded on the first signature, then overall verify succeeded, with debug enabled, only one successful signature info is printed out.
- if verification succeeded on the 4th signature, then overall verify succeeded, with debug enabled, the first 3 signatures with failed reason and the 4th successful signature are printed out
- if verification failed on all 50 signatures, then overall verify failed, with debug enabled, all the signatures with failed reasons are printed out. We also need to indicate the max number of 50 signatures are verified.

Reply:
> Let's keep it as it is for now. We can have a separate PR to resolve #201.

To summarize, keep it for now means we will have the following implementation:
1. Verification succeeded: the success outcome will be returned only. The verification process will stop at the first signature that gets verified successfully.
2. Verification failed: an empty verification outcome slice will be returned along with error `ErrorVerificationFailed`.